File: BaekJoon/DP/2565.py
import sys


N = int(input())

dp = []

for _ in range(N):
    a, b = map(int, sys.stdin.readline().split())
    dp.append((a,b))

dp.sort()

li = [1 for _ in range(N)]

for i in range(N):
    for j in range(i):
        if dp[i][1] > dp[j][1]:
            li[i] = max( li[i] , li[j]+1)

print(N-max(li))

File: BaekJoonSwift/BaekJoonSwift/DP (2)/2565.swift
import Foundation

let n = Int(readLine()!)!
var arr = [[Int]]()
var lis = [Int]()
for _ in 0..<n {
    arr.append(readLine()!.split(separator: " ").map { Int($0)! })
}

arr.sort { $0[0] < $1[0]}

for i in 0..<n {
    lis.append(1)
    for j in 0..<i {
        if arr[i][1] > arr[j][1] && lis[j] >= lis[i] {
            lis[i] = lis[j] + 1
        }
    }
}

print(n - lis.max()!)
